index.d.ts

```diff
@@ -0,0 +1,30 @@
+interface Point {
+  line: number;
+  column: number;
+  offset: number;
+}
+interface ASTNode {
```

Comment:
We could step it up a notch, and declare the node types and it's children. With that we can say `parser(text: string): Message` and it would contain all proper types/children per node 😄 I have a draft of that locally, [but it looks like this](https://commitist-7a7qx1zbs.vercel.app/spec/ast).

Reply:
@byCedric want to push against this branch, or alternatively open your own PR? The more detailed the better.

Reply:
I'll push it to your branch now

Reply:
We might need to update it a bit more, but you get the point 😄  Feel free to revert it